chore: Release v1.8.0 (#8655)
ZcashFoundation/zebra: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org).

> ## [Zebra 1.8.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.8.0) - 2024-07-01

Reviewer: ```suggestion
## [Zebra 1.8.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.8.0) - 2024-07-02
```

> ### Summary

Reviewer: Could we put the summary right after `Zebra 1.8.0` for consistency with past releases?

```suggestion
```

Author: I thought the "Summary" title was above `Zebra 1.8.0`, so I applied your suggestion and then added it back below `Zebra 1.8.0`. Only then did I realize you meant something else. Sorry. :D

Author: I removed it.

- Zebra now drops transactions with unpaid actions in block templates and from

>   the mempool.

Reviewer: Nitpick/Optional: It might be nice to re-word this so it's clear that the unpaid actions limit constant could still be changed relatively easily.

```suggestion
- Zebra now uses a default unpaid actions limit of 0, dropping transactions with any unpaid actions from the mempool and excluding them when selecting transactions from the mempool during block template construction.
```

- The `zebrad` binary no longer contains the scanner of shielded transactions.
  This means `zebrad` no longer contains users' viewing keys.
- Support for custom Testnets and Regtest is greatly enhanced.
- Windows is now back in the second tier of supported platforms.
- The end-of-support time interval is set to match `zcashd`'s 16 weeks.
- The RPC serialization of empty treestates matches `zcashd`. 

### Added

- Add an init function for a standalone `ReadStateService` ([#8595](https://github.com/ZcashFoundation/zebra/pull/8595))
- Add new parameters for custom Testnets ([#8477](https://github.com/ZcashFoundation/zebra/pull/8477), [#8518](https://github.com/ZcashFoundation/zebra/pull/8518), [#8524](https://github.com/ZcashFoundation/zebra/pull/8524))
- Add default constructions for several RPC method responses ([#8616](https://github.com/ZcashFoundation/zebra/pull/8616))

> - Allow custom Testnets to make peer connections and configure more parameters ([#8528](https://github.com/ZcashFoundation/zebra/pull/8528))

Reviewer: Optional: Could we move "Allow custom Testnets to make peer connections" to the "Fixed" section? It should've been doing that when custom Testnets were first added. Allowing configuration for more Testnet parameters should stay here, we could add the PR number to this first item.

Feel free to edit this suggestion.

```suggestion
- Allow configuring more parameters on custom Testnets and an NU5 activation height on Regtest ([#8477](https://github.com/ZcashFoundation/zebra/pull/8477), [#8518](https://github.com/ZcashFoundation/zebra/pull/8518), [#8524](https://github.com/ZcashFoundation/zebra/pull/8524), [#8528](https://github.com/ZcashFoundation/zebra/pull/8528), [#8636](https://github.com/ZcashFoundation/zebra/pull/8636))
- Add default constructions for several RPC method responses ([#8616](https://github.com/ZcashFoundation/zebra/pull/8616), [#8505](https://github.com/ZcashFoundation/zebra/pull/8505))
```

> - Allow configurable NU5 activation height on Regtest ([#8505](https://github.com/ZcashFoundation/zebra/pull/8505))

Reviewer: I think this could go under the "Fixed" section too, I had just broken it in https://github.com/ZcashFoundation/zebra/pull/8475, though in this PR, #8505, we did add support for constructing Canopy proposal blocks from block templates, and a small performance improvement, if we want to mention those here as well and keep it in this section.

```suggestion
- Support constructing Canopy proposal blocks from block templates ([#8505](https://github.com/ZcashFoundation/zebra/pull/8505))
```

#### Docs

- Add a section on Regtest usage to the Zebra book ([#8526](https://github.com/ZcashFoundation/zebra/pull/8526))

> - Describe custom Testnets in the Zebra book ([#8636](https://github.com/ZcashFoundation/zebra/pull/8636))

Reviewer: Optional/Nitpick: We could probably unify these entries.

```suggestion
- Document custom Testnets, Regtest and how they compare to Mainnet and the default Testnet in the Zebra book ([#8526](https://github.com/ZcashFoundation/zebra/pull/8526), [#8636](https://github.com/ZcashFoundation/zebra/pull/8636))
```

### Changed

> - Drop transactions with unpaid actions ([#8638](https://github.com/ZcashFoundation/zebra/pull/8638))

Reviewer: Could we mention that this is a result of changing the default unpaid action limit somewhere in this entry, maybe something like:

```suggestion
- Use a default unpaid action limit of 0 so that transactions with unpaid actions are excluded from the mempool and block templates by default ([#8638](https://github.com/ZcashFoundation/zebra/pull/8638))
```

> - Lower the mandatory checkpoint height ([#8629](https://github.com/ZcashFoundation/zebra/pull/8629))

Reviewer: Optional/Nitpick: We could add more context here, but this suggestion may be excessively verbose.

```suggestion
- Lower the mandatory checkpoint height from immediately above the ZIP-212 grace period to immediately below Canopy activation ([#8629](https://github.com/ZcashFoundation/zebra/pull/8629))
```

> - Use the new `zcash_script` callback API ([#8566](https://github.com/ZcashFoundation/zebra/pull/8566))

Reviewer: Optional/Nitpick: The impact here seems significant, it may be worthwhile to include context on it.

```suggestion
- Use the new `zcash_script` callback API in `zebra-script`, allowing Zebra's ECC dependencies to be upgraded independently of `zcash_script` and `zcashd` ([#8566](https://github.com/ZcashFoundation/zebra/pull/8566))
```

- Put Windows in Tier 2 of supported platforms ([#8637](https://github.com/ZcashFoundation/zebra/pull/8637))

> - Remove support for starting the scanner at `zebrad` startup ([#8594](https://github.com/ZcashFoundation/zebra/pull/8594))

Reviewer: I think we should mention that starting zebra-scan in zebrad was experimental and that we haven't removed support for any stable feature.

Optional/Nitpick: Should we include a note about the motivation here?

```suggestion
- Remove experimental support for starting the `zebra-scan` in the `zebrad` process as a step towards moving the scanner to its own process ([#8594](https://github.com/ZcashFoundation/zebra/pull/8594))
```

- Reduce the end of support time from 20 weeks to 16 weeks ([#8530](https://github.com/ZcashFoundation/zebra/pull/8530))

> - Restore parts of the internal-miner feature for use on Regtest ([#8506](https://github.com/ZcashFoundation/zebra/pull/8506))

Reviewer: ```suggestion
- Restore parts of the experimental `internal-miner` feature for use on Regtest ([#8506](https://github.com/ZcashFoundation/zebra/pull/8506))
```

### Fixed

> - Allow square brackets in network config listen addr and external addr ([#8504](https://github.com/ZcashFoundation/zebra/pull/8504))

Reviewer: Optional/Nitpick: This was working before as well but required users to explicitly define a port in their `listen_addr`

```suggestion
- Allow square brackets in network config listen addr and external addr without explicitly configuring a port in the listen addr ([#8504](https://github.com/ZcashFoundation/zebra/pull/8504))
```

Side note: I'm not sure if it's actually correct to define an IP address with the brackets and without a port, but, it's clear what the users mean. I'd want to log a warning when this happens in case the same structure doesn't work elsewhere (like in the RPC `listen_addr`), but I've had some difficulty using the `tracing` crate in the `serde` `Deserialization` impl, does anyone know why that doesn't seem to work?

- Fix general conditional compilation attributes ([#8602](https://github.com/ZcashFoundation/zebra/pull/8602))

> - Update `median_timespan()` method to align with zcashd implementation ([#8491](https://github.com/ZcashFoundation/zebra/pull/8491))

Reviewer: If we want to move allowing custom Testnets to make peer connections to the "Fixed" section here:

```suggestion
- Update `median_timespan()` method to align with zcashd implementation ([#8491](https://github.com/ZcashFoundation/zebra/pull/8491))
- Allow custom Testnets to make peer connections ([#8528](https://github.com/ZcashFoundation/zebra/pull/8528))
```

> - Refactor the serialization of empty treestates ([#8533](https://github.com/ZcashFoundation/zebra/pull/8533))

Reviewer: Could we mention that this was to fix a user-visible issue around empty note commitment trees, (and optionally that the new code is simpler and more maintainable), maybe something like:

```suggestion
- Refactor and simplify the serialization of empty treestates to fix an incorrect serialization of empty note commitment trees for RPCs ([#8533](https://github.com/ZcashFoundation/zebra/pull/8533))
```

> - Address port conflict issues and enable related Windows tests ([#8624](https://github.com/ZcashFoundation/zebra/pull/8624))

Reviewer: Optional: Could we highlight that the impact of the port conflict issue was limited to tests, and mostly impacted just Windows?

```suggestion
- Fix port conflict issue in some tests and re-enable those tests on Windows where those issues were especially problematic ([#8624](https://github.com/ZcashFoundation/zebra/pull/8624))
```

> - Fix a bug with trailing characters in the openapi spec method descriptions ([#8597](https://github.com/ZcashFoundation/zebra/pull/8597))

Reviewer: Nitpick:

```suggestion
- Fix a bug with trailing characters in the OpenAPI spec method descriptions ([#8597](https://github.com/ZcashFoundation/zebra/pull/8597))
```

> - Stop using `displaydoc` ([#8614](https://github.com/ZcashFoundation/zebra/pull/8614))

Reviewer: Optional: This one doesn't seem very visible for users, though it does remove a dependency, maybe that affects the binary size? (And the log messages were slightly modified).

```suggestion
```

Side note: It looks like https://github.com/yaahc/displaydoc/pull/47 has merged if we want to add this dependency back, it seems pretty convenient and might be useful when we work on https://github.com/ZcashFoundation/zebra/issues/2908 or other changes related to adding/modifying error types.
